temporalio/client.py

```diff
@@ -2830,6 +3097,130 @@ async def report_cancellation(
         )
 
 
+# TODO: This name is suboptimal now. We could deprecate it and introduce WorkflowActivityHandle as a
```

Comment:
Yeah, let's do that. Which means that you would have to accept a `WorkflowActivityHandle` where you accept `AsyncActivityHandle`.